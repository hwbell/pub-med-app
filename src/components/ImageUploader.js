import React from 'react';
import '../App.scss';
import 'bootstrap/dist/css/bootstrap.css';

// components
import { Button, Fade, Input } from 'reactstrap';

// ******************************************************************************
class ImageUploader extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedFile: null
    }

    this.onChangeHandler = this.onChangeHandler.bind(this);
    
  }

  onChangeHandler(event) {

    if(event) {
      this.setState({
        selectedFile: event.target.files[0],
        loaded: 0,
      }, () => {
        console.log(this.state.selectedFile)
      })
    }

  }

  render() {

    return (
      <div className="upload-container" style={{ alignSelf: 'flex-start' }}>
        <p style={{ fontSize: '14px' }}>add an image to your profile!</p>


        <Input className="form-control" bsSize="sm" 
          type="file" name="file" onChange={this.onChangeHandler}
          style={styles.input}>
        </Input>

        <Fade in={!!this.state.selectedFile}>
          <Button size="md" className="article-button add upload" style={styles.button}>
            upload pic
          </Button>
        </Fade>

      </div>
    );
  }
}

const styles = {

}

export default ImageUploader;
