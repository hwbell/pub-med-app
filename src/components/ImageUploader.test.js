import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ImageUploader from './ImageUploader';

describe('ImageUploader', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ImageUploader />);
  })

  it('renders without crashing', async () => {
    wrapper.update();
  });
  
  it('renders correctly', async () => {
    const tree = await renderer
      .create(<ImageUploader />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('contains the correct elements', () => {
    expect(wrapper.find('.upload-container').length).toBe(1);
    expect(wrapper.find('.form-control').length).toBe(1);
    expect(wrapper.find('Fade').length).toBe(1);
    expect(wrapper.find('Button').length).toBe(1);

    expect(wrapper.find('Fade').props().in).toBe(false);

  })

  it('should fire the this.onChange() when the user selects a file', () => {
    const onChangeSpy = jest.spyOn(ImageUploader.prototype, 'onChangeHandler');        
    let newWrapper = shallow(<ImageUploader />);

    newWrapper.find('.form-control').simulate('change');

    expect(onChangeSpy).toHaveBeenCalled();
  })
})
