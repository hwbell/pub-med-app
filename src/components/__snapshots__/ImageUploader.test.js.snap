// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUploader renders correctly 1`] = `
<div
  className="upload-container"
  style={
    Object {
      "alignSelf": "flex-start",
    }
  }
>
  <p
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    add an image to your profile!
  </p>
  <input
    className="form-control form-control-sm form-control-file"
    name="file"
    onChange={[Function]}
    type="file"
  />
  <div
    className="fade"
  >
    <button
      aria-label={null}
      className="article-button add upload btn btn-secondary btn-md"
      onClick={[Function]}
    >
      upload pic
    </button>
  </div>
</div>
`;
